lambdas/data-migration2/tests/test-pdrs.js

```diff
@@ -0,0 +1,336 @@
+  await recursivelyDeleteS3Bucket(process.env.system_bucket);
+
+  await providersModel.deleteTable();
+  await collectionsModel.deleteTable();
+  await pdrsModel.deleteTable();
+
+  await t.context.knex.destroy();
+  await t.context.knexAdmin.raw(`drop database if exists "${testDbName}"`);
+  await t.context.knexAdmin.destroy();
+});
+
+test.serial('migratePdrRecord correctly migrates PDR record', async (t) => {
+  const { knex, testCollection, testProvider, collectionCumulusId, providerCumulusId } = t.context;
+
+  const testPdr = generateTestPdr(testCollection.name, testProvider.id);
+  await migratePdrRecord(testPdr, knex);
+
+  const record = await t.context.knex.queryBuilder()
+    .select()
+    .table(tableNames.pdrs)
+    .where({ name: testPdr.pdrName })
+    .first();
+
+  t.like(
+    omit(record, ['cumulus_id']),
+    omit({
+      name: testPdr.pdrName,
+      provider_cumulus_id: providerCumulusId,
+      collection_cumulus_id: collectionCumulusId,
+      execution_cumulus_id: testPdr.executionCumulusId ? testPdr.executionCumulusId : null,
+      status: testPdr.status,
+      progress: testPdr.progress,
+      pan_sent: testPdr.PANSent,
+      pan_message: testPdr.PANmessage,
+      stats: testPdr.stats,
+      address: testPdr.address,
+      original_url: testPdr.originalUrl,
+      timestamp: new Date(testPdr.timestamp),
+      duration: testPdr.duration,
+      created_at: new Date(testPdr.createdAt),
+      updated_at: new Date(testPdr.updatedAt),
+    },
+    ['updated_at'])
+  );
+});
+
+test.serial('migratePdrRecord throws SchemaValidationError on invalid source data from DynamoDB', async (t) => {
+  const { knex, testCollection, testProvider } = t.context;
+
+  const testPdr = generateTestPdr(testCollection.name, testProvider.id);
+
+  delete testPdr.status;
+
+  await t.throwsAsync(
+    migratePdrRecord(testPdr, knex),
+    { name: 'SchemaValidationError' }
+  );
+});
+
+test.serial('migratePdrRecord handles nullable fields on source PDR data', async (t) => {
+  const { knex, testCollection, testProvider, collectionCumulusId, providerCumulusId } = t.context;
+
+  const testPdr = generateTestPdr(testCollection.name, testProvider.id);
+
+  delete testPdr.execution;
+  delete testPdr.PANSent;
+  delete testPdr.PANmessage;
+  delete testPdr.progress;
+  delete testPdr.stats;
+  delete testPdr.address;
+  delete testPdr.originalUrl;
+  delete testPdr.timestamp;
+  delete testPdr.duration;
+  delete testPdr.updatedAt;
+
+  await migratePdrRecord(testPdr, knex);
+  const record = await t.context.knex.queryBuilder()
+    .select()
+    .table(tableNames.pdrs)
+    .where({ name: testPdr.pdrName })
+    .first();
+
+  t.like(
+    omit(record, ['cumulus_id']),
+    omit({
+      ...testPdr,
+      name: testPdr.pdrName,
+      provider_cumulus_id: providerCumulusId,
+      collection_cumulus_id: collectionCumulusId,
+      execution_cumulus_id: testPdr.executionCumulusId ? testPdr.executionCumulusId : null,
```

Comment:
It seems like this should always be null based on the test setup?

Reply:
https://github.com/nasa/cumulus/pull/2034#discussion_r562376483